### PR TITLE
TST: Add test for filling new rows through reindexing MultiIndex

### LINKED ISSUE #23693: DataFrame.reindex with specified fill method fails for MultiIndex
#### Code Sample, a copy-pastable example if possible

```python
import pandas as pd

i = pd.MultiIndex.from_tuples([('a', 'b'), ('d', 'e')])
df = pd.DataFrame([[0, 7], [3, 4]], index=i, columns=['x', 'y'])

print(df)
#      x  y
# a b  0  7
# d e  3  4

i2 = pd.MultiIndex.from_tuples([('a', 'b'), ('d', 'e'), ('h', 'i')])
# same behavior
#i2 = pd.MultiIndex.from_tuples([('a', 'b', 'c'), ('d', 'e', 'f'), ('h', 'i', 'j')])

print(df.reindex(i2, axis=0, method='ffill'))
#       x    y
# a b  3.0  4.0
# d e  NaN  NaN
# h i  0.0  7.0

```
#### Problem description

The reindexing operation above introduces a row to the `MultiIndex`. When no fill method is specified the new row is added and filled with NA as expected.

When `ffill` is specified the behavior is not explainable for me. The index is updated as expected but a NA row is added in the middle of the existing data.

#### Expected Output

Not sure if `ffill` for MultiIndexes is designed like this, but I was hoping for

```
#       x    y
# a b  3.0  4.0
# d e  0.0  7.0
# h i  0.0  7.0
```

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit: None

pandas: 0.23.4
pytest: 3.7.1
pip: 18.1
setuptools: 39.0.1
Cython: None
numpy: 1.15.0
scipy: None
pyarrow: 0.11.1
xarray: None
IPython: 6.5.0
sphinx: None
patsy: None
dateutil: 2.7.3
pytz: 2018.5
blosc: 1.6.1
bottleneck: 1.2.1
tables: None
numexpr: 2.6.8
feather: None
matplotlib: 3.0.2
openpyxl: 2.5.5
xlrd: 1.1.0
xlwt: None
xlsxwriter: None
lxml: None
bs4: None
html5lib: 1.0.1
sqlalchemy: 1.2.10
pymysql: None
psycopg2: 2.7.5 (dt dec pq3 ext lo64)
jinja2: 2.10
s3fs: None
fastparquet: 0.1.6
pandas_gbq: None
pandas_datareader: None
</details>

### PR DESCRIPTION
**Pull request by phofl**
- [x] closes #23693
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

Issue was fixed previously

### CODE REVIEW

File: pandas/tests/indexes/multi/test_reindex.py [hunk 1]
```diff
@@ -101,3 +101,13 @@ def test_reindex_non_unique():
     msg = "cannot handle a non-unique multi-index!"
     with pytest.raises(ValueError, match=msg):
         a.reindex(new_idx)
+
+
+def test_reindex_fill_added_rows():
+    # GH: 23693
+    i = pd.MultiIndex.from_tuples([("a", "b"), ("d", "e")])
```
Comment (jbrockmendel) [inline pandas/tests/indexes/multi/test_reindex.py:108]: can you use MultiIndex with out the "pd"

Comment (jbrockmendel) [reply pandas/tests/indexes/multi/test_reindex.py:108]: and pls avoid 1-letter variable names.  `mi` is a good choice for MultiIndex obj

File: pandas/tests/indexes/multi/test_reindex.py [hunk 2]
```diff
@@ -101,3 +101,13 @@ def test_reindex_non_unique():
...
+    df = pd.DataFrame([[0, 7], [3, 4]], index=i, columns=["x", "y"])
+    i2 = pd.MultiIndex.from_tuples([("a", "b"), ("d", "e"), ("h", "i")])
+    result = df.reindex(i2, axis=0, method="ffill")
+    expected = pd.DataFrame([[0, 7], [3, 4], [3, 4]], index=i2, columns=["x", "y"])
```
Comment (jbrockmendel) [inline pandas/tests/indexes/multi/test_reindex.py:112]: if the method being tested is DataFrame.reindex, then this belongs in tests.frame.method.test_reindex

File: pandas/tests/indexes/multi/test_reindex.py (hunk 1 above)
Comment (phofl) [reply pandas/tests/indexes/multi/test_reindex.py]: Done

File: pandas/tests/indexes/multi/test_reindex.py (hunk 2 above)
Comment (phofl) [reply pandas/tests/indexes/multi/test_reindex.py]: Sorry, was checking frame/indexing and could not find a reindex file. Moved the test now to the methods folder

File: pandas/tests/frame/methods/test_reindex.py [hunk 3]
```diff
@@ -864,3 +864,12 @@ def test_reindex_signature(self):
             "fill_value",
             "tolerance",
         }
+
+    def test_reindex_fill_added_rows(self):
```
Comment (jbrockmendel) [inline pandas/tests/frame/methods/test_reindex.py:868]: can you add "multiindex" in the name somehow, and "fill" -> "ffill"

half-sentence after the GH ref below on what the original problem was?

and total nitpick exclusively if you want to do me a kindness: I like the pattern `# GH#23693` for the GH references

Comment (phofl) [reply pandas/tests/frame/methods/test_reindex.py]: All done :)

**Comment by jreback** [comment]
thanks @phofl

**Pull request merged**